src/app/cache.service.ts

```diff
@@ -39,7 +39,12 @@ export class CacheService {
     } else if (fallback && fallback instanceof Observable) {
       this.inFlightObservables.set(key, new Subject());
       console.log(`%c Calling api for ${key}`, 'color: purple');
-      return fallback.do((value) => { this.set(key, value, maxAge); });
+      return fallback
+        .do((value) => { this.set(key, value, maxAge); })
+        .catch((err) => { 
+          this.notifyInFlightObservers(key, null); 
```

Comment:
would be good if the observers get the error thrown, since they could individually handle any errors from the backend.